### PR TITLE
feat/qr scanner

### PR DESCRIPTION
**Pull request by boompttn**

### CODE REVIEW

File: pages/login/index.tsx [hunk 1]
```diff
@@ -27,30 +32,73 @@ const Home: React.FC = observer(() => {
...
+        }}
       />
+      <p className={`text-red-600 pt-2 ${!isOpen && 'hidden'}`}>
+        Please enter 10 digits
+      </p>
 
       <p className='mt-10'>or Scan QR</p>
 
-      <label className='mt-3 py-2 px-8 bg-gray-800 text-white rounded cursor-pointer'>
+      {/* <label className='mt-3 py-2 px-8 bg-gray-800 text-white rounded cursor-pointer'>
```
Comment (new5558) [inline pages/login/index.tsx]: remove comments

File: pages/login/index.tsx [hunk 2]
```diff
@@ -27,30 +32,73 @@ const Home: React.FC = observer(() => {
...
+
+      <button
+        onClick={() => setQrReady(true)}
+        type='button'
+        className='mt-3 py-2 px-8 bg-gray-800 text-white rounded'
+      >
+        Open Camera
+      </button>
+
+      {qrReady ? (
```
Comment (new5558) [inline pages/login/index.tsx]: change to qrReady  && <QrReader
          className='w-40 h-40 mt-4 rounded'
          delay={300}
          onError={handleError}
          onScan={handleScan}
        />

File: pages/login/index.tsx [hunk 3]
```diff
@@ -27,30 +32,73 @@ const Home: React.FC = observer(() => {
...
     <form onSubmit={login} className='flex flex-col items-center'>
       <img
         className='mt-16 border-black border-solid border w-40 h-40 rounded'
-        src='https://images.pexels.com/photos/3496994/pexels-photo-3496994.jpeg?cs=srgb&dl=white-and-black-2020-with-confetti-3496994.jpg&fm=jpg'
+        src='images/icon.png'
         alt='logo'
       />
       <p className='mt-8'>
         Please enter your <span className='underline'>Ticket ID</span>
       </p>
```
Comment (new5558) [inline pages/login/index.tsx]: install prettier and enable vscode set format on save.

File: pages/login/index.tsx (hunk 1 above)
Comment (new5558) [inline pages/login/index.tsx]: remove this

File: pages/login/index.tsx [hunk 4]
```diff
@@ -2,6 +2,8 @@
 import React, { useCallback, useContext, useState } from 'react';
 import { observer } from 'mobx-react-lite';
 import { useRouter } from 'next/router';
+// import QrReader from 'react-qr-reader';
```
Comment (new5558) [inline pages/login/index.tsx]: remove this

File: pages/login/index.tsx [hunk 5]
```diff
@@ -27,30 +32,73 @@ const Home: React.FC = observer(() => {
     [ticketID]
   );
 
+  const handleScan = (data: string | null) => {
```
Comment (new5558) [inline pages/login/index.tsx:35]: consider using  [useCallback](https://reactjs.org/docs/hooks-reference.html#usecallback)

File: pages/login/index.tsx [hunk 6]
```diff
@@ -27,30 +32,73 @@ const Home: React.FC = observer(() => {
...
+      setTicketID(data);
+      setQrReady(false)
+    }
+  };
+
+  const handleError = (err: any) => {
+    console.error(err);
+  };
+
+  const QrReader = dynamic(() => import('react-qr-reader'), {
```
Comment (new5558) [inline pages/login/index.tsx]: consider moving this component outside functional component to avoid rerender

File: pages/login/index.tsx [hunk 7]
```diff
@@ -27,30 +32,73 @@ const Home: React.FC = observer(() => {
...
-        className='mt-5 border-black border-solid border rounded'
+        className='mt-5 border-black border-solid border rounded p-1'
         type='text'
         value={ticketID}
-        onChange={e => setTicketID(e.target.value)}
+        onChange={e => {
+          const _ticketID = e.target.value;
+          setTicketID(_ticketID);
+          if (_ticketID.length < 10 || _ticketID.length > 10) {
+            setIsOpen(true);
```
Comment (new5558) [inline pages/login/index.tsx]: try changing name to setLoginError, loginError

File: pages/login/index.tsx (hunk 6 above)
Comment (new5558) [inline pages/login/index.tsx]: Also try using react suspense to do [lazyload](https://blog.bitsrc.io/lazy-loading-react-components-with-react-lazy-and-suspense-f05c4cfde10c)

File: pages/login/index.tsx [hunk 8]
```diff
@@ -12,7 +14,10 @@ const Home: React.FC = observer(() => {
   const { userStore } = useContext(rootContext);
   const [error, setError] = useState<string | null>(null);
   const router = useRouter();
-  const [ticketID, setTicketID] = useState();
+  const [isOpen, setIsOpen] = useState(false);
```
Comment (new5558) [inline pages/login/index.tsx]: useState<boolean>

**Review by new5558 — CHANGES_REQUESTED** [review]

**Review by hamzaabamboo — COMMENTED** [review]
LGTM

File: pages/login/index.tsx (hunk 2 above)
Comment (boompttn) [reply pages/login/index.tsx]: && with what?

File: pages/login/index.tsx (hunk 8 above)
Comment (boompttn) [reply pages/login/index.tsx]: What do you mean? it's already useState

**Review by new5558 — APPROVED** [review]

**Pull request merged**